vm/src/types/program.rs

```diff
@@ -162,6 +162,50 @@ impl Program {
             builtins,
         })
     }
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_for_proof(
+        builtins: Vec<BuiltinName>,
+        data: Vec<MaybeRelocatable>,
+        start: Option<usize>,
+        end: Option<usize>,
```

Comment:
Aren't this mandatory for proof mode? If so, instead of passing `Option<usize>` I think it makes sense to just pass `usize` and use `Some(start)` and `Some(end)` when initializing the struct.

Reply:
Makes sense